File: frontend/src/types/Project.ts
// the template for the project object, which will be used to type the data we get from the backend
export interface Project {
    projectId: number;
    projectName: string;
    projectType: string;
    projectRegionalProgram: string;
    projectImpact: number;
    projectPhase: string;
    projectFunctionalityStatus: string;
}

File: backend/WaterProject.API/Data/Project.cs
using System.ComponentModel.DataAnnotations;

namespace WaterProject.API.Data
{
    public class Project // what one record (project) looks like in the Projects table in the DB
    {
        [Key]
        public int ProjectId { get; set; }
        [Required]
        public string? ProjectName { get; set; }
        public string? ProjectType { get; set; }
        public string? ProjectRegionalProgram { get; set; }
        public int? ProjectImpact { get; set; }
        public string? ProjectPhase { get; set; }
        public string? ProjectFunctionalityStatus { get; set; }

    }
}
